aldryn_events/tests/frontend/integration/pages/alteli.js

```diff
@@ -0,0 +1,166 @@
+             * specify directory)
+             */
+            writeScreenShot: function (data, filename) {
+                var stream = fs.createWriteStream(
+                    screenShotDirectory + filename
+                );
+
+                stream.write(new Buffer(data, 'base64'));
+                stream.end();
+            }
+        };
+
+        jasmine.getEnv().addReporter({
+            specDone: function (result) {
+                browser.getCapabilities().then(function (capabilities) {
+                    browser.takeScreenshot().then(function (png) {
+                        var browserName = capabilities.caps_.browserName;
+                        var filename =
+                        browserName + '-' +
+                        result.status.toUpperCase() + '-' +
+                        result.fullName.replace(/ /g, '_') + '.png';
+
+                        Utils.writeScreenShot(png, filename);
+                    });
+                });
+            }
+        });
+    },
+
+    // utility to test expectation an element is missing
+    expectByCssToBeAbsent: function (element) {
+        element.isPresent().then(function (present) {
+            expect(present).toBeFalsy();
+        });
+    },
+
+    setEmail: function (emailInput, email) {
+        emailInput.clear().then(function () {
+            emailInput.sendKeys(email);
+        });
+    },
+    setPassword: function (passwordInput, password) {
+        passwordInput.clear().then(function () {
+            passwordInput.sendKeys(password);
+        });
+    },
+
+    loginToSite: function (
+        emailInput,
+        email,
+        passwordInput,
+        password,
+        loginButton,
+        userMenu
+        ) {
+        emailInput.clear().then(function () {
+            // fill in email field
+            emailInput.sendKeys(email).then(function () {
+                passwordInput.clear().then(function () {
+                    // fill in password field
+                    passwordInput.sendKeys(password).then(function () {
+                        loginButton.click();
+
+                        browser.wait(function () {
+                            return browser.isElementPresent(userMenu);
+                        }, 8000);
+
+                        // validate user menu
+                        expect(userMenu.isDisplayed()).toBeTruthy();
+                    });
+                });
+            });
+        });
+    },
+
+    switchToNewTab: function (newTabURL) {
+        browser.getAllWindowHandles().then(function (handles) {
+            var newWindowHandle = handles[1];
+
+            // put the focus on the second tab
+            return browser.switchTo().window(newWindowHandle);
+        }).then(function () {
+            // validate new tab url
+            expect(browser.driver.getCurrentUrl()).toContain(newTabURL);
+        });
+    },
+
+    switchToOldTab: function (closeOldTab) {
```

Comment:
i do not really like the boolean trap here. if the user is not familiar with the lib - you don't know what `switchToOldTab(true)` does. it might be slightly better to do an options param so you could call it like this
`switchToOldTab({ closeOldTab: true })`